beacon_node/network/src/sync/block_sidecar_coupling.rs

```diff
@@ -300,82 +315,66 @@ impl<E: EthSpec> RangeBlockComponentsRequest<E> {
                 .insert(index, column)
                 .is_some()
             {
-                return Err(CouplingError {
-                    msg: format!("Repeated column block_root {block_root:?} index {index}"),
-                    column_and_peer: None,
-                });
+                return Err(CouplingError::InternalError(format!(
+                    "Repeated column block_root {block_root:?} index {index}"
+                )));
             }
         }
 
         // Now iterate all blocks ensuring that the block roots of each block and data column match,
         // plus we have columns for our custody requirements
         let mut rpc_blocks = Vec::with_capacity(blocks.len());
 
+        let exceeded_retries = attempt >= MAX_COLUMN_RETRIES;
         for block in blocks {
             let block_root = get_block_root(&block);
             rpc_blocks.push(if block.num_expected_blobs() > 0 {
                 let Some(mut data_columns_by_index) = data_columns_by_block.remove(&block_root)
                 else {
-                    // This PR ignores the fix from https://github.com/sigp/lighthouse/pull/5675
-                    // which allows blobs to not match blocks.
-                    // TODO(das): on the initial version of PeerDAS the beacon chain does not check
-                    // rpc custody requirements and dropping this check can allow the block to have
-                    // an inconsistent DB.
-
-                    // For now, we always assume that the block peer is right.
-                    // This is potentially dangerous as we can get isolated on a chain with a
-                    // malicious block peer.
-                    // TODO: fix this by checking the proposer signature before downloading columns.
                     let responsible_peers = column_to_peer.iter().map(|c| (*c.0, *c.1)).collect();
-                    return Err(CouplingError {
-                        msg: format!("No columns for block {block_root:?} with data"),
-                        column_and_peer: Some((responsible_peers, PeerAction::LowToleranceError)),
+                    return Err(CouplingError::PeerFailure {
```

Comment:
I wonder if it makes sense to accumulate the errors and `continue` the for loop instead?

Reply:
I tried to keep changes minimal here. Besides, if we just collect the error, we can't use `data_columns_by_index` below so we'll have to fail there. Its a bit inefficient, but it works

Reply:
I think we could still `continue` to pair the next block, but i guess if the peers' response is missing one block, we'd have to retry fetching columns for all blocks in the range anyways, so it's likely not very useful to accumulate the errors.